### PR TITLE
handling modular equations in solveset

### LINKED ISSUE #13178: Ability to solve modular equations with integer solutions
See for instance https://www.johndcook.com/blog/2017/08/22/predicting-a-lcg-output/. It would be cool if you could just write

```py
n = symbols('n', integer=True)
a = 742938285
z = 1898888478
m = 2**31 - 1
x = 20170816
solveset(x – Mod(a**n*z, m), n, S.Integers)
```

and it produced the answer {100}.

### PR DESCRIPTION
**Pull request by sharma-kunal**
fixes #13178
This PR is built over PR #14284. The `_inverse_modular` was almost complete but it still had some issues which I tried to remove in this PR. 

#### References to other Issues or PRs

```
n = symbols('n', integer=True)
a = 742938285
z = 1898888478
m = 2**31 - 1
x = 20170816
f = x - Mod(a**n*z, m)
>>> solveset(f, n, S.Integers) 
{100}
```

ping @asmeurer @smichr @normalhuman @Yathartha22
#### Release Notes

* solvers
   * implemented `_solve_modular` for solving modular equations

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This has failed the code quality test with:
```
AssertionError: File ends with more than 1 newline: /home/travis/build/sympy/sympy/sympy/solvers/solveset.py, line 3112
```

You can see that error by clicking the detail link below to see the Travis log. You can also run `$bin/test quality` locally to see that before pushing. If you fix that then the rest of the tests can run.

**Comment by sharma-kunal** [comment]
Sorry for the late response, my exams were going on. I'll just check for the quality test and update the PR.

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -326,6 +326,52 @@ def _invert_complex(f, g_ys, symbol):
     return (f, g_ys)
```
Comment (sharma-kunal) [inline sympy/solvers/solveset.py]: I don't know how but the `_invert_abs` function was somehow missing. Maybe overlapped by `invert_modular` function. So I just copied the function from my master branch and pasted it here. No changes have been done to the function `_invert_abs`.

**Comment by czgdp1807** [comment]
The issue associated with this PR has been fixed. So closing this PR. Thanks for the contribution though.

**Pull request closed without merging**